bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -32,15 +32,13 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
 
   /** Syntax for ObsContext. */
   implicit class ObsContextOps(ctx: ObsContext) {
-    def isEligibleForBags: Boolean = {
-      val targets = ctx.getTargets
-      val prgs    = targets.getGuideEnvironment.getPrimaryReferencedGuiders
-      prgs.isEmpty || prgs.asScala.exists { gp =>
-        targets.getPrimaryGuideProbeTargets(gp).asScalaOpt.exists(_.getBagsResult match {
-          case BagsResult.NoSearchPerformed => true
-          case _                            => false
-        })
-      }
+    // Check the primary guide group to see if there are any manual targets. If there are, do not run BAGS.
+    def hasManualPrimary: Boolean = {
+      ctx.getTargets.getGuideEnvironment.getPrimary.asScalaOpt.exists(_.getAll.asScalaList.exists(_.primaryIsManual))
+    }
+
+    def isMissingSearch: Boolean = {
+      ctx.getTargets.getGuideEnvironment.getPrimary.asScalaOpt.forall(_.getAll.asScalaList.exists(_.getBagsResult == BagsResult.NoSearchPerformed))
     }
   }
```

Comment:
Sure. I guess since it was just two boolean functions differing in the existential modifier that it didn't initially occur to me or seem important. I've gone ahead and refactored in my recent commit as suggested.